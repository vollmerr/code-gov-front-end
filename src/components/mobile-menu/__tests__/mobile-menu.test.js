import React from 'react';

import Menu from 'components/mobile-menu'

import { shallow } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <Menu /> component', () => {
  const props = {
    open: false,
    menu: [
      {name: "test-name", url: "https://test-url.com"},
      {name: "test-name2", url: "/test-url2"}
    ],
    dispatch: jest.fn()
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Menu store={store} {...props}/>);
  });
  it('renders Menu', () => {
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  })
  it('renders Menu component with open prop', () => {
    props.open=true;
    const store = mockStore(props);
    wrapper = shallow(<Menu store={store} {...props}/>).dive();
    expect(wrapper.exists()).toEqual(true);
  });
    it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})