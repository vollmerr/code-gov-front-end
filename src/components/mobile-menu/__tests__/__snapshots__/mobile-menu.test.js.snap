// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Menu /> component should render correctly 1`] = `
<Menu
  dispatch={[Function]}
  menu={
    Array [
      Object {
        "name": "test-name",
        "url": "https://test-url.com",
      },
      Object {
        "name": "test-name2",
        "url": "/test-url2",
      },
    ]
  }
  open={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
